test/e2e/tests/test_model_package.py

```diff
@@ -0,0 +1,276 @@
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_model_package_resource_status,
+            reference,
+        )
+
+    def _wait_sagemaker_model_package_status(
+        self,
+        model_package_name,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_model_package_sagemaker_status,
+            model_package_name,
+        )
+
+    def _assert_model_package_status_in_sync(
+        self, model_package_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_model_package_status(
+                model_package_name, expected_status
+            )
+            == self._wait_resource_model_package_status(reference, expected_status)
+            == expected_status
+        )
+
+    def test_unversioned_model_package_completed(
+        self, xgboost_unversioned_model_package
+    ):
+        (reference, resource) = xgboost_unversioned_model_package
+        assert k8s.get_resource_exists(reference)
+
+        model_package_name = resource["spec"].get("modelPackageName", None)
+        assert model_package_name is not None
+
+        model_package_desc = get_sagemaker_model_package(model_package_name)
+        model_package_arn = model_package_desc["ModelPackageArn"]
+
+        if k8s.get_resource_arn(resource) is None:
+            logging.error(
+                f"ARN for this resource is None, resource status is: {resource['status']}"
+            )
+        
+        assert k8s.get_resource_arn(resource) == model_package_arn
+
+        resource_tags = resource["spec"].get("tags", None)
+        assert_tags_in_sync(model_package_arn, resource_tags)
+
+        assert model_package_desc["ModelPackageStatus"] == cfg.JOB_STATUS_INPROGRESS
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        self._assert_model_package_status_in_sync(
+            model_package_name, reference, cfg.JOB_STATUS_COMPLETED
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
+
+        # Check that you can delete a completed resource from k8s
+        _, deleted = k8s.delete_custom_resource(reference, 3, 10)
+        assert deleted is True
+        assert (
+            get_sagemaker_model_package(model_package_name)
+            is None
+        )
+
+
+    def test_versioned_model_package_completed(self, xgboost_versioned_model_package):
+        (reference, spec, resource) = xgboost_versioned_model_package
+        assert k8s.get_resource_exists(reference)
+
+        if resource["spec"].get("modelPackageGroupName") is not None:
```

Comment:
the objective of assert at 233 is to check from the service side, so recommend using `list_model_packages()` and filter by PackageGroupName. This should contain only one item because of name randomisation and use the ARN from that for the assertion